src/compiler/builder.ts

```diff
@@ -1170,19 +1230,26 @@ namespace ts {
                             const file = sourceFiles[0];
                             const info = state.fileInfos.get(file.resolvedPath)!;
                             if (info.signature === file.version) {
-                                newSignature = computeSignature(text, data, computeHash);
-                                if (newSignature !== file.version) { // Update it
+                                const signature = computeSignatureWithDiagnostics(
+                                    state.program!,
+                                    text,
+                                    computeHash,
+                                    getCanonicalFileName,
+                                    data,
+                                );
+                                if (!data?.diagnostics?.length) newSignature = signature;

```

Comment:
Why is this conditional?

Reply:
I have renamed it to emitSignature and added comment but essence of it is, the signature of file with this change is d.ts emit text + declaration diagnostics while we want to store emitSigature = d.ts emit text in build info to figure out when d.ts file changes between compilations